src/api/app/views/webui/request/beta_show_tabs/_review_summary.html.haml

```diff
@@ -1,28 +1,27 @@
--# FIXME: We don't render anything when the project got deleted because this nullifies the association. It should point to
-  `Project.deleted_instance` or maybe the project got created with the same name again in the meanwhile?
-- unless (review.for_project? || review.for_package?) && review.project.nil?
-  .d-flex.flex-row.pt-1
-    .pl-3
-      - if review.accepted?
-        %i.fas.fa-sm.fa-check.text-primary.align-middle.pr-1
-      - if review.declined?
-        %i.fas.fa-times.text-danger.align-middle.pr-2
-      - if review.new?
-        %i.fas.fa-2xs.fa-circle.text-warning.align-middle.pr-2
+:ruby
+  deleted_project = (review.for_project? || review.for_package?) && review.project.nil?
+  deleted_package = review.for_package? && review.package.nil?
 
-    = render BsRequestOverviewAvatarsComponent.new(review)
+.d-flex.flex-row.pt-1
+  .pl-3
+    - if review.accepted?
+      %i.fas.fa-sm.fa-check.text-primary.align-middle.pr-1
+    - if review.declined?
+      %i.fas.fa-times.text-danger.align-middle.pr-2
+    - if review.new?
+      %i.fas.fa-2xs.fa-circle.text-warning.align-middle.pr-2
 
-    .pl-1
-      - if review.for_user?
-        = user_with_realname_and_icon(review.by_user, short: true, no_icon: true)
-      - elsif review.for_group?
-        = link_to(review.by_group, group_path(Group.find_by(title: review.by_group)))
-      - elsif review.for_package?
-        = link_to("#{review.by_project} / #{review.by_package}", package_users_path(project: review.by_project,
-          package: review.by_package))
-      - elsif review.for_project?
-        = link_to(review.by_project, project_users_path(project: review.by_project))
+  = render BsRequestOverviewAvatarsComponent.new(review) unless deleted_project || deleted_package
```

Comment:
The logic which is checking if this view component must be rendered or not should be inside the view component itself. This is possible with [render?](https://github.com/openSUSE/open-build-service/wiki/View-Components#conditional-rendering). Otherwise, this logic needs to be duplicated whenever we reuse this component.

Reply:
Thanks for the reminder! Have another look, please.